src/business_logic/transaction/objects/internal_declare.rs

```diff
@@ -0,0 +1,428 @@
+            fee::{calculate_tx_fee, execute_fee_transfer, FeeInfo},
+        },
+    },
+    core::{
+        contract_address::starknet_contract_address::compute_class_hash,
+        transaction_hash::starknet_transaction_hash::calculate_declare_transaction_hash,
+    },
+    definitions::{general_config::StarknetGeneralConfig, transaction_type::TransactionType},
+    services::api::contract_class::{ContractClass, EntryPointType},
+    utils::{calculate_tx_resources, felt_to_hash, verify_no_calls_to_other_contracts, Address},
+};
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+///  Represents an internal transaction in the StarkNet network that is a declaration of a Cairo
+///  contract class.
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+pub(crate) struct InternalDeclare {
+    pub(crate) class_hash: [u8; 32],
+    pub(crate) sender_address: Address,
+    pub(crate) tx_type: TransactionType,
+    pub(crate) validate_entry_point_selector: Felt,
+    pub(crate) version: u64,
+    pub(crate) max_fee: u64,
+    pub(crate) signature: Vec<Felt>,
+    pub(crate) nonce: Felt,
+    pub(crate) hash_value: Felt,
+}
+
+// ------------------------------------------------------------
+//                        Functions
+// ------------------------------------------------------------
+impl InternalDeclare {
+    #![allow(unused)] // TODO: delete once used
+    pub fn new(
+        contract_class: ContractClass,
+        chain_id: Felt,
+        sender_address: Address,
+        max_fee: u64,
+        version: u64,
+        signature: Vec<Felt>,
+        nonce: Felt,
+    ) -> Result<Self, TransactionError> {
+        let hash = compute_class_hash(&contract_class)?;
+        let class_hash = felt_to_hash(&hash);
+
+        let hash_value = calculate_declare_transaction_hash(
+            contract_class,
+            chain_id,
+            sender_address.clone(),
+            max_fee,
+            version,
+            nonce.clone(),
+        )?;
+        // Value generated from get_selector_from_name(VALIDATE_DECLARE_ENTRY_POINT_NAME)
+        let validate_entry_point_selector = felt_str!(
+            "1148189391774113786911959041662034419554430000171893651982484995704491697075"
+        );
+
+        let internal_declare = InternalDeclare {
+            class_hash,
+            sender_address,
+            tx_type: TransactionType::Declare,
+            validate_entry_point_selector,
+            version,
+            max_fee,
+            signature,
+            nonce,
+            hash_value,
+        };
+
+        internal_declare.verify_version()?;
+
+        Ok(internal_declare)
+    }
+
+    pub fn account_contract_address(&self) -> Address {
+        self.sender_address.clone()
+    }
+
+    pub fn get_calldata(&self) -> Vec<Felt> {
+        let bytes = Felt::from_bytes_be(&self.class_hash);
+        Vec::from([bytes])
+    }
```

Comment:
This looks like a getter.
```suggestion
    pub fn get_calldata(&self) -> &[Felt] {
        &[Felt::from_bytes_be(&self.class_hash)]
    }
```